### ISSUE TITLE
Combining different fields and indices fails

### ISSUE THREAD

**Issue by pdnooteboom** [OceanParcels/parcels#436]
If one combines 3D and 2D fields and makes use of indices, only the surface velocity currents are used to advect particles.  A quick fix is to add a line indices = indices.copy() in field.py:

diff --git a/parcels/field.py b/parcels/field.py
index 163c325..bb67c86 100644
--- a/parcels/field.py
+++ b/parcels/field.py
@@ -273,19 +273,32 @@ class Field(object):
                full_load is however sometimes necessary for plotting the fields.
         """
 
+        indices = indices.copy()
         if not isinstance(filenames, Iterable) or isinstance(filenames, str):
             filenames = [filenames]
         dimension_filename = dimension_filename if dimension_filename else filenames[0]
         if indices is None:
             indices = {}
         with NetcdfFileBuffer(dimension_filename, dimensions, indices) as filebuffer:

**Comment by erikvansebille** [comment]
See also PR #437

**Comment by erikvansebille** [comment]
Wouldn't another solution to this Issue be to change line 191 in fieldset.py:
```
inds = indices[var] if (indices and var in indices) else indices
```
The problem here is that if `indices` is a dictionary but `var` is not in that dictionary, that `inds` then becomes the whole dictionary. That doesn't seem right. 

Should it be something like
```
if type(indices) is dict:
    inds = indices[var].copy() if var in indices else None
else:
    inds = indices.copy()
```

**Comment by delandmeterp** [comment]
I don't understand.
A typical indices is a dictionary:
`indices = {'lon': indslon, 'lat': indslat}`

What line 191 does it to process indices like:
`indices = {'U':{'lon': indslonU, 'lat': indslatU}, 'V':{'lon':indslonV, 'lat':indslatV} }`

So indices is always a dictionary, no? `var` in line 191 is not 'lon', or 'lat' (the dimensions) but the fields names. So this proposed solution couldn't work if I'm not mistaken

**Comment by erikvansebille** [comment]
Fixed in #437
